slatedb/src/dispatcher.rs

```diff
@@ -210,8 +217,10 @@ impl<T: Send + std::fmt::Debug> MessageDispatcher<T> {
     /// A [Result] containing the [DbState::error] if it was already set, or the result
     /// of [MessageDispatcher::handle_result] otherwise.
     pub(crate) async fn run(&mut self) -> Result<(), SlateDBError> {
-        // TODO: handle panic here (similar to spawn_bg_task)
-        let result = self.run_loop().await;
+        let result = AssertUnwindSafe(self.run_loop())
+            .catch_unwind()
+            .await
+            .unwrap_or_else(Self::handle_panic);
         let result = self.handle_result(result);
         if let Err(e) = self.cleanup(result.clone()).await {
```

Comment:
I'm not sure if we want to catch panics in `cleanup` as well. We don't in `spawn_bg_task`.